ocaml/xapi/db_gc_util.ml

```diff
@@ -246,6 +252,24 @@ let gc_certificates ~__context =
      )
   |> List.iter (fun (cert, _) -> Db.Certificate.destroy ~__context ~self:cert)
 
+let gc_vtpms ~__context =
+  let all_vtpms = Db.VTPM.get_all ~__context in
+  let vm_vtpms =
+    Seq.concat_map
+      (fun vm -> Db.VM.get_VTPMs ~__context ~self:vm |> List.to_seq)
+      (Db.VM.get_all ~__context |> List.to_seq)
+    |> VtpmSet.of_seq
+  in
+  all_vtpms
```

Comment:
Could `DB.VTPM.get_VM` be used to retrieve the VM ref, check whether the Ref is valid and has a backlink to the vtpm.